### DISCUSSION TITLE
How to know if script command is already run and recording the terminal session?

### DISCUSSION THREAD

**Discussion by ma08** [util-linux/util-linux#1768] [Q&A] [answered]
I found in [this stackoverflow answer](https://stackoverflow.com/a/59671817/3465519) that `script` supposedly adds a commandline variable `SCRIPT`. Yet, when I tried it myself, it turned out to be empty.

Any ideas on how to check if `script` command is already running?

**Comment by karelzak** [comment, marked as answer]
The implementation from util-linux does not set any env. variable, but maybe it's a good idea to add this feature to script(1). I'll think about it.

The most robust is probably to use something like flock to protect the session for the recursive script(1) calls. 

The simple solution is to check the process parents, for example:
```
$ pstree -s $$ 
systemd───systemd───gnome-terminal-───bash───script───bash───pstree
```

**Reply by ma08** [reply]
Thanks for your response! @karelzak 

I would love to contribute if you decide to implement the feature. Please let me know.

**Comment by karelzak** [comment]
Go ahead and contribute! ;-) I think the feature is a good idea.

Maybe we can export more information (by env.vars) to the session, like SCRIPT_PID=, SCRIPT_CPMMAND=, etc.

**Reply by ma08** [reply]
Awesome. I will get started on that in my free time. I am pretty new to contributing to a project like this. So will take my time to go through everything before I start. Would appreciate any inputs :)

What is the best place to communicate with the developers in case I have any questions? Is the IRC channel active?

**Reply by ma08** [reply]
@karelzak 

There is a guide for contributing at [https://kzak.redcrew.org/doku.php?id=util-linux:start](https://kzak.redcrew.org/doku.php?id=util-linux:start). Is that upto date?

**Reply by ma08** [reply]
Also this [Documentation/howto-contribute.txt](https://github.com/util-linux/util-linux/blob/master/Documentation/howto-contribute.txt).

**Comment by karelzak** [comment]
Yes, https://github.com/util-linux/util-linux/blob/master/Documentation/howto-contribute.txt is a good starting point; see also https://github.com/util-linux/util-linux/blob/master/README where is a note about IRC.

**Reply by ma08** [reply]
Thank you. Will get started from here.
